### PR TITLE
add shared library support

### PR DESCRIPTION
**Pull request by lnslbrty**
Signed-off-by: Toni Uhlig <matzeton@googlemail.com>

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,9 +1,9 @@
-project(facil.io C)
+project(facilio C)
```
Comment (boazsegev) [inline CMakeLists.txt]: Is there a reason to drop the `.`?

If so, I'd like to know.

File: CMakeLists.txt [hunk 2]
```diff
@@ -42,3 +42,6 @@ target_include_directories(facil.io
   PUBLIC  lib/facil/redis
 )
 
+set_property(TARGET facilio_objlib PROPERTY POSITION_INDEPENDENT_CODE 1)
```
Comment (boazsegev) [inline CMakeLists.txt]: Interesting... I'm no CMake user / expert - what do these lines do?

Does this replace the option of using facil.io as a source code library?

File: CMakeLists.txt (hunk 1 above)
Comment (lnslbrty) [reply CMakeLists.txt]: It is confusing (and probably not best-practice) using a `.` for library names.
This character is usually reserved as separator between library name, library suffix and version.

File: CMakeLists.txt (hunk 2 above)
Comment (lnslbrty) [reply CMakeLists.txt]: It basically adds a `CFLAG` named `-fPIC` which tells the compiler to generate position independent code.
For shared libraries this is a must as the start address of the memory mapped library can be different to the default image base address set by the ELF linker.

**Comment by lnslbrty** [comment]
> I assume that adding shared / static library support does not remove the ability to use facil.io as a code library?

It won't. There is still a static library named `libfacilio_static.a` (using `*_static.a` is a common design pattern for CMake based projects).

**Pull request closed without merging**